app/src/main/java/org/simple/clinic/user/LoggedInUserHttpInterceptor.kt

```diff
@@ -12,16 +13,20 @@ class LoggedInUserHttpInterceptor @Inject constructor() : Interceptor {
 
   override fun intercept(chain: Interceptor.Chain?): Response {
     val originalRequest = chain!!.request()
-    val updatedRequest = originalRequest.newBuilder()
 
-    if (userSession.isUserLoggedIn()) {
-      val (loggedInUser) = userSession.loggedInUser().blockingFirst()
-      val (accessToken) = userSession.accessToken()
+    val (user) = userSession.loggedInUser().blockingFirst()
+    val (accessToken) = userSession.accessToken()
 
-      updatedRequest.addHeader("Authorization", "Bearer ${accessToken!!}")
-      updatedRequest.addHeader("X-USER-ID", loggedInUser!!.uuid.toString())
+    return when (user?.isApprovedForSyncing() == true) {
```

Comment:
Can we replace this with an `if-else` expression?

Reply:
TODO: Separate discussion about if-else vs when.